projects/aca-content/src/lib/components/search/search-results/search-results.component.ts

```diff
@@ -207,42 +204,55 @@ export class SearchResultsComponent extends PageComponent implements OnInit {
     this.columns = this.extensions.documentListPresets.searchResults || [];
 
     if (this.route) {
-      this.route.queryParams.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((params: Params) => {
-        this.savedSearchesService
-          .getSavedSearches()
-          .pipe(takeUntilDestroyed(this.destroyRef))
-          .subscribe((savedSearches) => {
-            const savedSearchFound = savedSearches.find((savedSearch) => savedSearch.encodedUrl === encodeURIComponent(params[this.queryParamName]));
-            this.initialSavedSearch = savedSearchFound !== undefined ? savedSearchFound : this.initialSavedSearch;
-          });
-        if (params[this.queryParamName]) {
-          this.isLoading = true;
-        }
-        this.loadedFilters$.next();
-        this.encodedQuery = params[this.queryParamName] || null;
-        this.searchedWord = extractSearchedWordFromEncodedQuery(this.encodedQuery);
-        this.updateUserQuery();
-        const filtersFromEncodedQuery = extractFiltersFromEncodedQuery(this.encodedQuery);
-        if (filtersFromEncodedQuery !== null) {
-          const filtersToLoad = this.queryBuilder.categories.length;
-          let loadedFilters = this.searchedWord === '' ? 0 : 1;
-          this.queryBuilder.filterLoaded
-            .asObservable()
-            .pipe(takeUntilDestroyed(this.destroyRef), takeUntil(this.loadedFilters$))
-            .subscribe(() => {
+      combineLatest([
+        this.route.queryParams,
+        this.router.events.pipe(
+          filter((e): e is NavigationStart => e instanceof NavigationStart),
+          startWith(null)
+        )
+      ])
+        .pipe(takeUntilDestroyed(this.destroyRef))
+        .subscribe(([params, navigationStartEvent]) => {
+          const shouldExecuteQuery = this.shouldExecuteQuery(navigationStartEvent, params);
+
+          this.savedSearchesService
+            .getSavedSearches()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((savedSearches) => {
+              const savedSearchFound = savedSearches.find(
+                (savedSearch) => savedSearch.encodedUrl === encodeURIComponent(params[this.queryParamName])
+              );
+              this.initialSavedSearch = savedSearchFound !== undefined ? savedSearchFound : this.initialSavedSearch;
+            });
+          if (params[this.queryParamName]) {
+            this.isLoading = true;
+          }
+          this.loadedFilters$.next();
+          this.encodedQuery = params[this.queryParamName] || null;
```

Comment:
Probably || null can be skipped because if it finds nothing in params then it will give undefined and undefined similarly like null is falsy so should work similarly